File: api/src/Workshop.Api/Data/Entities/AccessRequest.cs
namespace Workshop.Api.Data.Entities;

public class AccessRequest
{
    public int Id { get; set; }
    public int ReservableObjectId { get; set; }
    public string RequesterUsername { get; set; } = string.Empty;
    public DateTime RequestedStartTime { get; set; }
    public DateTime RequestedEndTime { get; set; }
    public string Status { get; set; } = "Pending"; // Pending, Approved, Denied
    public string? Message { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? RespondedAt { get; set; }
    public string? ResponseMessage { get; set; }
    
    public ReservableObject ReservableObject { get; set; } = null!;
}

File: client/src/app/models/access-request.ts
export interface AccessRequest {
  id: number;
  reservableObjectId: number;
  reservableObjectName: string;
  requesterUsername: string;
  requestedStartTime: string;
  requestedEndTime: string;
  status: AccessRequestStatus;
  message?: string;
  responseMessage?: string;
  createdAt: string;
  respondedAt?: string;
}

export type AccessRequestStatus = 'Pending' | 'Approved' | 'Denied';

export interface CreateAccessRequestRequest {
  reservableObjectId: number;
  requestedStartTime: string;
  requestedEndTime: string;
  message?: string;
}

export interface RespondToAccessRequestRequest {
  status: 'Approved' | 'Denied';
  responseMessage?: string;
}
